File: stubs/java/src/main/java/com/beastmode/rmddd/examples/ecommerce/OrderCreatedEvent.java
package com.beastmode.rmddd.examples.ecommerce;

import com.beastmode.rmddd.domain.DomainEvent;

import java.util.Map;

/**
 * Domain event fired when an order is created.
 */
public class OrderCreatedEvent extends DomainEvent {
    
    private final CustomerId customerId;
    
    public OrderCreatedEvent(OrderId orderId, CustomerId customerId) {
        super(orderId);
        this.customerId = customerId;
    }
    
    public OrderId getOrderId() {
        return (OrderId) getAggregateId();
    }
    
    public CustomerId getCustomerId() {
        return customerId;
    }
    
    @Override
    public Map<String, Object> getEventData() {
        return Map.of(
            "order_id", getOrderId().getValue().toString(),
            "customer_id", customerId.getValue().toString()
        );
    }
}

File: stubs/csharp/src/BeastMode.RmDdd.Examples/Examples/ECommerce/Events/OrderCreatedEvent.cs
using System.Collections.Generic;
using BeastMode.RmDdd.Domain;

namespace BeastMode.RmDdd.Examples.ECommerce.Events
{
    /// <summary>
    /// Domain event fired when an order is created.
    /// </summary>
    public class OrderCreatedEvent : DomainEvent
    {
        public CustomerId CustomerId { get; }
        
        public OrderCreatedEvent(OrderId orderId, CustomerId customerId) : base(orderId)
        {
            CustomerId = customerId;
        }
        
        public OrderId OrderId => (OrderId)AggregateId;
        
        public override IDictionary<string, object> GetEventData()
        {
            return new Dictionary<string, object>
            {
                ["order_id"] = OrderId.Value.ToString(),
                ["customer_id"] = CustomerId.Value.ToString()
            };
        }
    }
}